app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,66 @@
+class Api::V1::UsersController < ApplicationController
+  before_action :set_user, only: %i[ show update destroy ]
+  include Authentication
+
+
+  # GET /users
+  def index
+    render json: {working: "yes"}, status: 200
+    if authenticate_user
+
+      users = User.all
+
+      render json: users
+    else
+      render status: 401
+    end
```

Comment:
will give double render error if `authenticate_user` is true because you already rendered a json above

Reply:
Fixed it.